tests/admin/controller/test_library.py

```diff
@@ -444,3 +491,86 @@ def test_library_delete(self):
 
         library = get_one(self._db, Library, uuid=library.uuid)
         eq_(None, library)
+
+    def test_find_location_through_registry(self):
+        library = self._default_library
+        controller = self.manager.admin_library_settings_controller
+        original_ask_registry = controller.ask_registry
+        get = self.do_request
+        test = self
+        class Mock(LibrarySettingsController):
+            called_with = []
+            def mock_ask_registry(self, service_area_object):
```

Comment:
You should be able to override `ask_registry` rather than defining a method with a different name and assigning it to `ask_registry`.